beetle/iroh-bitswap/src/lib.rs

```diff
@@ -333,9 +330,7 @@ impl<S: Store> Bitswap<S> {
                     *entry.get_mut() = new_state;
                 }
                 match new_state {
-                    PeerState::DialFailure(_)
-                    | PeerState::Disconnected
-                    | PeerState::Unresponsive => {
+                    PeerState::Disconnected | PeerState::Unresponsive => {
```

Comment:
Do we expect to ever see DialFailure here?

Reply:
No the DialFailure variant has been removed. Its no longer a valid reason to change the peer state, so its no longer a peer state.